doc/man7/provider.pod

```diff
@@ -72,6 +72,39 @@ the provider.
 Note that the provider will not be made available for applications to use until
 the initialization function has completed and returned successfully.
 
+I<< If the provider is a shared module and uses F<libcrypto> functions,
+it is B<strongly> recommended that the initialization function initialises
+F<libcrypto> appropriately, by including the following lines as early as
+possible: >>
+
+ if ((settings = OPENSSL_INIT_new()) == NULL
+     || !OPENSSL_INIT_set_upcalls(settings, handle, in)
+     || !OPENSSL_init_crypto(OPENSSL_INIT_PROVIDER, settings)) {
+     OPENSSL_INIT_free(settings);
+     return 0;
+ }
+ OPENSSL_INIT_free(settings);
+
+Those lines can be in providers that are statically linked into the application
+as well.  They are not necessary in that case, but are harmless, so they can be
+used unconditionally in source code that is used equally for providers that are
+made into dynamically loadable modules and for providers that are built into the
+application.
+
+=begin comment
```

Comment:
This shouldn't be in a comment.  A comment is a guarantee that nobody will ever read it.  The text should be included in the man page or removed entirely.  In the latter case, they could be included in a C source file as a comment.

I think they contain useful information and could be included here.

Reply:
That information doesn't really concern external people that want to write a provider.

So, where would you place those comments in code?  I've tried to figure out where I would be most likely to go looking if I was wondering, and this is what I came up with, se re "nobody will ever read it", I'm apparently nobody...

Reply:
I may have found another way, so fix coming up

Reply:
Done